PyGamer_Thermal_Camera/code.py

```diff
@@ -0,0 +1,431 @@
+
+    histo_bucket = [0, 0, 0, 0, 0, 0, 0, 0]  # Clear histogram bucket
+    for row in range(7, -1, -1):  # Collect camera data and calculate spectrum
+        for col in range(0, 8):
+            value = map_range(image[col][row],
+                              MIN_SENSOR_C, MAX_SENSOR_C,
+                              MIN_SENSOR_C, MAX_SENSOR_C)
+            histo_index = int(map_range(value, MIN_RANGE_C, MAX_RANGE_C, 0, 7))
+            histo_bucket[histo_index] = histo_bucket[histo_index] + 1
+            sum_bucket = sum_bucket + value  # Calculate sum for average
+            minimum = min(value, minimum)
+            maximum = max(value, maximum)
+
+    for col in range(0, 8):  # Display histogram
+        for row in range(0, 8):
+            if histo_bucket[col] / 8 > 7 - row:
+                image_group[((row * 8) + col) + 1].fill = element_color[col]
+            else:
+                image_group[((row * 8) + col) + 1].fill = BLACK
+    return minimum, maximum, sum_bucket
+
+def setup_mode():  # Set alarm threshold and minimum/maximum range values
+    status_label.color = WHITE
+    status_label.text  = "-SET-"
+
+    ave_label.color = BLACK  # Turn off average label and value display
+    ave_value.color = BLACK
+
+    max_value.text = str(MAX_RANGE_F)  # Display maximum range value
+    min_value.text = str(MIN_RANGE_F)  # Display minimum range value
+
+    time.sleep(0.8)  # Show SET status text before setting parameters
+    status_label.text  = ""  # Clear status text
+
+    param_index = 0  # Reset index of parameter to set
+
+    # Select parameter to set
+    while not panel.button.start:
+        while (not panel.button.a) and (not panel.button.start):
+            left, right, up, down = move_buttons(joystick=panel.has_joystick)
+            if up:
+                param_index = param_index - 1
+            if down:
+                param_index = param_index + 1
+            if param_index > 2:
+                param_index = 2
+            if param_index < 0:
+                param_index = 0
+            status_label.text = param_list[param_index][0]
+            image_group[param_index + 66].color = BLACK
+            status_label.color = BLACK
+            time.sleep(0.2)
+            image_group[param_index + 66].color = param_list[param_index][1]
+            status_label.color = WHITE
+            time.sleep(0.2)
+
+        if panel.button.a:  # Button A pressed
+            panel.play_tone(1319, 0.030)  # E6
+        while panel.button.a:  # wait for button release
+            pass
+
+        # Adjust parameter value
+        param_value = int(image_group[param_index + 70].text)
+        while (not panel.button.a) and (not panel.button.start):
+            left, right, up, down = move_buttons(joystick=panel.has_joystick)
+            if up:
+                param_value = param_value + 1
+            if down:
+                param_value = param_value - 1
+            if param_value  > MAX_SENSOR_F:
```

Comment:
could do param_value = max(MIN_SENSOR_F, min(MAX_SENSOR_F, param_value))

Reply:
Excellent simplification. Thank you!

Reply:
Could also do that at line 170, as well.